portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -28,24 +36,60 @@
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
-public class DataServlet extends HttpServlet {
-  private ArrayList<String> trial_array;
-  private String json_array;
-
+public final class DataServlet extends HttpServlet {
+  private ArrayList<Task> tasks;
   @Override
-  public void init() {
-    trial_array = new ArrayList<>();
-    trial_array.add("First hard-coded message");
-    trial_array.add("Second hard-coded message");
-    trial_array.add("Third hard-coded message");
-    json_array = new Gson().toJson(trial_array);
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String comment = request.getParameter("comment-box");
+    String displayName = request.getParameter("display-name");
```

Comment:
We can avoid accessing this parameter if anonymous is set to true. I'd move this line in an else block corresponding to the if statement on line 48.

Reply:
will do in one of the later branches that are open